### PR TITLE
CI/CD Process for Re-Ignited-Phone

### PR DESCRIPTION
**Pull request by gabrielcalderon**
* Added support for building artifacts for both ESX 1.1 and 1.2 using Github Actions.
* Updated Node CI stage to use *actions/setup-node@v2-beta* which has built-in caching for faster builds.
* Completely refactored the Readme file.
* Added badges for CI, Release Tag, Patreon, Discord, License
* Update Instructions file.
* The CI process will **NOT** run if changes to SQL and .MD files are made.
* Simplifies the process of creating a new tag, Just download the latest Resources from the latest Master build and create a tag with those.

You can see how the new readme looks [Here](https://github.com/gabrielcalderon/Re-Ignited-Phone/)

Note:

* With the new CI process the users are no longer required to move server.lua files and modify their config when using ESX 1.2. For every Push/Pull Request Github will build and upload artifacts for each ESX version.

* Basically the CI will take care of the manual file moving and setup required for each ESX version.

* The CI process will create two artifacts: **resources-esx-1-1** and **resources-esx-1-2**

### CODE REVIEW

**Comment by DmACKGL** [comment]
For what I'm reading, If I push a pull request at any it's going to publish a release automatically without actually merging it?

**Comment by gabrielcalderon** [comment]
@DmACKGL The current CI process only builds the code to make sure that there's no errors. It does not create any downloadable artifact.

The current process for installing Re-Ignited-Phone requires the user to move files depending on their version of ESX. My PR fixes this by created two separate zip files, one for each ESX version.

If you go to the build for this pull-request you can see that it create **two** zip-files, if you download them, they are ready to be used in a live server, without requiring the user to move server.lua files, or changing the config for esx 1.2. 

Build for this PR is [**here**](https://github.com/Re-Ignited-Development/Re-Ignited-Phone/actions/runs/222426773)

## Benefits
* Using my process, each push to master, and each Pull-Request will get resource zip files auto-generated that can be used for testing in a live server. 
* Eliminates the need for the end-user to move server.lua files and change the config. They will only need to download the appropriate ZIP file for their ESX version.
* When creating a new tag, from Master, you will be able to download the latest resource zip files for the master branch directly from github without having to build your own. These two zip files can be used to create a tag/release directly in gitlab, even from your phone.
